server/routes/incidentRole/incidentRolePage.ts

```diff
@@ -197,24 +191,11 @@ export default class IncidentRolePage {
       const removeExistingOffences = incidentRoleChanged
       await this.saveToApiUpdate(postValues.draftId, incidentDetailsToSave, removeExistingOffences, user as User)
 
-      let defaultNextPage = NextPageSelectionAfterEdit.TASK_LIST
-      if (this.pageOptions.isPreviouslySubmitted()) {
-        defaultNextPage = NextPageSelectionAfterEdit.OFFENCE_DETAILS
-      }
-      const nextPageChoice = chooseNextPageAfterEdit(defaultNextPage, incidentRoleChanged)
-      switch (nextPageChoice) {
-        case NextPageSelectionAfterEdit.OFFENCE_SELECTION:
-          return redirectToOffenceSelection(res, postValues.draftId, incidentDetailsToSave.currentIncidentRoleSelection)
-        case NextPageSelectionAfterEdit.OFFENCE_DETAILS:
-          return redirectToOffenceDetails(res, postValues.draftId)
-        default:
-        // Fall through
-      }
-      if (this.pageOptions.isPreviouslySubmitted()) {
-        return redirectToTaskList(res, postValues.draftId)
+      const offencesExist = !removeExistingOffences && offenceDetails?.length > 0
+      if (!!req.session.forceOffenceSelection || !offencesExist) {
```

Comment:
So there are 3 reasons for going to the offence selection page:

 - The saved draft doesn't have any offences
 - The incident role has changed (and therefore the offences will be deleted)
 - This page is been shown as part of the "Re-add first offence" journey